plugins/inputs/beat/beat.go

```diff
@@ -0,0 +1,268 @@
+package beat
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	jsonparser "github.com/influxdata/telegraf/plugins/parsers/json"
+)
+
+const sampleConfig = `
+  ## An URL from which to read Beat-formatted JSON
+  ## Default is "http://127.0.0.1:5066".
+  url = "http://127.0.0.1:5066"
+
+  ## Enable collection of the Beat stats
+  # collect_beat_stats = true
+
+  ## Enable the collection if Libbeat stats
+  # collect_libbeat_stats = true
+
+  ## Enable the collection of OS level stats
+  # collect_system_stats = false
+
+  ## Enable the collection of Filebeat stats
+  # collect_filebeat_stats = true
```

Comment:
Concerning our "default value" discussion below, it might be easier to allow the user to specify a list, i.e.
```
collect =["beat", "libbeat", "filebeat"]
```
This way you know if an option was selected (it's in the list) or not. Maybe you can define an empty list to "collect all"...

Reply:
This is also more extensible in the future if you want to e.g. increase granularity...

Reply:
I don't have a strong opinion on this, but don't feel much motivation to change it either. The design decision to use booleans predated my commits which sought to integrate the plugins in the simplest way possible. The use of booleans to toggle functionality seemed similar to other plugins so it didn't seem unusual to use them in this way. I don't see how it would make a difference to the end user either, but I obviously have a limited perspective on this.

Reply:
Yeah it's similar to the plugins where we currently have a lot of trouble extending them without breaking configs... Many plugins therefore move to `include`, `exclude` lists to allow sophisticated filtering of metrics. This is nice for the end-user for cases where you want all but one.

Reply:
So is this just a "nice to have" or a blocker for moving forward? I can see how it'd be useful for some more complicated use cases or extensibility, but I don't have much reason to believe that how it is currently implemented doesn't solve the problems of the present. And I'm reluctant to spend much time solving the problems of the future, when they can be easily addressed via a PR once they materialize.

I'd assume that the prospective users of this plugin would prefer having a potentially less-perfect solution sooner than a more future proof solution later, but I defer to anyone else who is following this thread to chime in with their use cases and preferences, as all I have to go by on is my personal one.

Reply:
I would vote for nice to have vs blocker. i know i would love to see this out so i can monitor filebeats status even if its not perfect. Although if its not too much work it may be reasonable to do it now since i know fixing things later rarely happens and although i would love to contribute i have 0 knowledge of GO to even pick this up myself in a second PR.

Perhaps merging this functional PR as is with the non perfect config but then opening an issue or second PR to enhance the config may be a compromise between both options.

Reply:
The problem is that we **cannot** fix this *later* as we need to keep backward compatible with the config. Instead of arguing over this again I implemented what is necessary in less time than the discussion took so far...
Please see my comment below with the attached patch (`git apply ...`). Feel free to use it!

Reply:
Thanks, applied.